src/coverlet.core/Instrumentation/Instrumenter.cs

```diff
@@ -157,21 +185,40 @@ private Instruction AddInstrumentationCode(MethodDefinition method, ILProcessor
             return pathInstr;
         }
 
-        private static bool IsBranchTarget(ILProcessor processor, Instruction instruction)
+        private Instruction AddInstrumentationCode(MethodDefinition method, ILProcessor processor, Instruction instruction, BranchPoint branchPoint)
         {
-            foreach (var _instruction in processor.Body.Instructions)
+            var document = _result.Documents.FirstOrDefault(d => d.Path == branchPoint.Document);
+            if (document == null)
             {
-                if (_instruction.Operand is Instruction target)
-                {
-                    if (target == instruction)
-                        return true;
-                }
-
-                if (_instruction.Operand is Instruction[] targets)
-                    return targets.Any(t => t == instruction);
+                document = new Document { Path = branchPoint.Document };
+                _result.Documents.Add(document);
             }
 
-            return false;
+            if (!document.Branches.Exists(l => l.Number == branchPoint.StartLine && l.Ordinal == branchPoint.Ordinal))
+                document.Branches.Add(
+                    new Branch
+                    {
+                        Number = branchPoint.StartLine,
+                        Class = method.DeclaringType.FullName,
+                        Method = method.FullName,
+                        Offset = branchPoint.Offset,
+                        EndOffset = branchPoint.EndOffset,
+                        Path = branchPoint.Path,
+                        Ordinal = branchPoint.Ordinal
+                    }
+                );
+
+            string marker = $"B,{document.Path},{branchPoint.StartLine},{branchPoint.Ordinal}";
+
+            var pathInstr = Instruction.Create(OpCodes.Ldstr, _result.HitsFilePath);
+            var markInstr = Instruction.Create(OpCodes.Ldstr, marker);
+            var callInstr = Instruction.Create(OpCodes.Call, processor.Body.Method.Module.ImportReference(typeof(CoverageTracker).GetMethod("MarkExecuted")));
```

Comment:
Can we change the `ImportReference()` call to use the one-off `_markExecutedMethodLoader.Value` too? Other than that this looks good!